### PR TITLE
Removed hover effects and fixed mobile nav menus' toggle buttons

### LINKED ISSUE #4444: Mobile Nav Menus fail to close
Clicking the account component and other menus in the header after opening them in mobile view doesn't close them.

This issue is an extension to #4200. Re-clicking the account component and the _My Books_ and _Browse_ menus in the header, after opening them, doesn't close them. Ideally they should be closed after re-clicking them.   

### Relevant url?

https://openlibrary.org/

### Steps to Reproduce

1. Go to https://openlibrary.org/
2. Open Developer Console, toggle to Mobile View and click on the _Browse_ or the _My Books_ menu to open either of them
3. Click on the opened menu again to close it.

To reproduce the bug for the account component:
1. Go to https://openlibrary.org/ and login
2. Open Developer Console, toggle to Mobile View and click on the account component:

3. Click on the it again to close it.

* Actual: Step 3 doesn't do anything.
* Expected: Step 3 should close the menu.

### Details

- **Logged in (Y/N)?** Yes for the account component, No for the others
- **Browser type/version?** Firefox Version 84.0.2 (64-bit)
- **Operating system?** Ubuntu 20.04
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

A similar solution as has been used for #4200 should fix the isssue

### PR DESCRIPTION
**Pull request by shrey27tri01**
### Issue Number
Closes #4444 

### What does this PR achieve? 
Fix

### Technical

I have used the [checkbox hack](https://css-tricks.com/the-checkbox-hack/) to fix the toggle buttons of the mobile navbar menus. Also, removed the hover effect from the buttons, since they led to a not-so-friendly user interface.

### Testing

1. Open OpenLibrary in development mode
2. Open the developer console and toggle to mobile view
3. Open the Browse menu/More options menu/Account dropdown
4. Click on it again to close it. 

### Screencast

I'm attaching a screencast for how it looks right now:

https://user-images.githubusercontent.com/52366674/106347861-a6c7a000-62e7-11eb-96fb-06e7cc33b7b0.mp4

### CODE REVIEW

**Comment by jdlrobson** [comment]
Sorry for the delay - will get to this one by the end of the week!

**Comment by shrey27tri01** [comment]
No problem! :)

File: openlibrary/templates/lib/nav_head.html [hunk 1]
```diff
@@ -77,8 +81,11 @@
       </div>
     </li>
     <li class="more-menu">
-      <a>$_('More') <img class="down-arrow"
-        width="7" height="4" aria-hidden="true" alt="" role="presentation" src="/static/images/down-arrow.png"/></a>
+      <label role="button" for="toggle-more-menu" class="more-menu-button">$_('More')
```
Comment (jdlrobson) [inline openlibrary/templates/lib/nav_head.html:84]: Suggested follow up: Given the repetition could we use a macro in the templates for each of these?

File: openlibrary/templates/lib/nav_head.html [hunk 2]
```diff
@@ -65,7 +65,11 @@
 
   <ul class="navigation-component">
     <li class="browse-menu">
-      <a>$_('Browse') <img class="down-arrow" width="7" height="4" aria-hidden="true" src="/static/images/down-arrow.png" alt="" role="presentation"/></a>
+      <label role="button" for="toggle-browse-menu" class="browse-menu-button">$_('Browse')
```
Comment (jdlrobson) [inline openlibrary/templates/lib/nav_head.html:68]: Recently we discovered the w3c validator [1] doesn't like the role here, it needs to be on the checkbox itself.

[1] https://validator.w3.org/nu/#textarea

**Review by jdlrobson — CHANGES_REQUESTED** [review]
Screen Shot 2021-02-03 at 10 19 06 PM
It's possible to open all the menus at the same time which doesn't seem right to me. Perhaps its possible to use a radio input so that only one can be open at a time?

**Comment by jdlrobson** [comment]
@shrey27tri01 do my requests make sense here or can I help clarify them some  how?

**Comment by shrey27tri01** [comment]
Really sorry, couldn't work on it. I will try and let you know if I need more clarification!

**Pull request merged**